features/0351-generic-json-message/README.md

```diff
@@ -0,0 +1,114 @@
+
+#### Threading
+
+This protocol supports but does not require the use of the `~thread` decorator to provide message threading.
+
+## Reference
+
+#### Message Format
+
+This protocol defines one DIDComm message format, of type `https://didcomm.org/json/1.0/message`. A `message` only requires (in addition to the standard contents of DIDComm messages) a `data-type` field and some number of JSON attachments, under the optional fields `data`, `data~attach`, and `~attach`. 
+
+The `"data-type"` field indicates the type of data contained under the attachments.
+
+The attachment fields are all optional, though there should be at least one.
+
+```json
+{
+  "@id": "123456780",
+  "@type": "https://didcomm.org/json/1.0/message",
+  "data-type": "",
+  "data": {},
+  "data~attach": {},
+  "~attach": []
+}
+```
+
+##### `data-type`
+
+The sender must provide a `data-type` identification string constant, which indicates, for the receiver, how the data is to be processed, and indicates, for the receiver-agent, which receiver(s) to send the message to.
+
+The receiver registers on a particular data-type value with the receiver-agent, in which case the receiver-agent must relay any message with that data-type value to the receiver. Multiple receivers can register with a single receiver-agent on a single data-type value, and a receiver can register with multiple receiver-agents on a data-type value.
+
+The receiver must be aware of the `data-type` value of a received message, whether by registering a specific handler for a specific `data-type` value, or by being sent the `data-type` value in addition to the content.
+
+##### `data`, `data~attach`, `~attach`
+
+These 3 fields are optional, although at least one must be present. They carry attachments (inline, embedded, and appended, respectively).
+
+The payload transported in this message is placed in an [attachment](/concepts/0017-attachments/README.md). If the attachment is embedded or appended, the receiver is provided with the entire attachment descriptor.
+
+The receiver-agent must be able to process all three fields.
+
+#### Threading
+
+A protocol built on top of generic JSON messages can include [`~thread` decorators](/concepts/0008-message-id-and-threading/README.md) in the messages. For example, this allows the receiver-agent to relay a reply from the receiver, so the receiver is able to reply to a message.
+
+For threading with senders or receivers that aren't aries agents, the sender-agent or receiver-agent must maintain context to correlate the DIDComm message thread, and the message thread of the communication protocol with the non-aries communicator. The lifetime of this context, for a custom protocol built on top of generic-JSON-message, is defined by the custom protocol.
```

Comment:
Timing decorators could also be useful for the message lifetimes.
... but that's a generic statement :).

Reply:
I mentioned timing decorators in HTTP over DIDComm, I could move that here, make this protocol describe an optional mechanism for supporting duplex communication, and in HTTP over DIDComm refer to this mechanism for the request-response nature of the protocol.

Reply:
Makes sense. I think you should retain a summary in the HTTP one as well.